[MRG] Updated adjusted_mutual_info_score and normalized_mutual_info_score default to 'arithmetic'

scouvreur opened the pull request:
#### Reference Issues/PRs

This was raised by @glemaitre in issue #13797 as issues to resolve before the v0.22 release.

#### What does this implement/fix? Explain your changes.

In `adjusted_mutual_info_score()` and ` normalized_mutual_info_score()` function from `sklearn.metrics.cluster`:
- [x] changed default to `arithmetic`
- [x] removed deprecation warning
- [x] removed logical structure for when `warn` is passed

In test for both functions:
- [x] updated expected returned values
- [x] removed `test_future_warning()`altogether

In other parts of the codebase calling either functions:
- [x] Removed explicit passing of `average_method='arithmetic'`

scouvreur committed:
Updated adjusted_mutual_info_score to have default as 'arithmetic' as specified in docstring, and updated corresponding test

scouvreur committed:
Updated expected value in test causing crash

glemaitre commented:
Thanks.

Could you make `git grep adjusted_mutual_info_score` and change the example and potential tests where we explicitly forced `average_method='arithmetic'`. Now, we don't need it and it could be make the examples simpler.

scouvreur commented:
Hi @glemaitre, using `git grep adjusted_mutual_info_score` I could not find any place in the docs or the tests where `average_method='arithmetic'` is passed.

Do you have any idea how I could maintain similar code coverage on this diff ?

glemaitre commented on sklearn/metrics/cluster/supervised.py:
```diff
@@ -740,11 +739,7 @@ def adjusted_mutual_info_score(labels_true, labels_pred,
 
     """
     if average_method == 'warn':
```
you should remove this line

glemaitre commented on sklearn/metrics/cluster/supervised.py:
```diff
@@ -740,11 +739,7 @@ def adjusted_mutual_info_score(labels_true, labels_pred,
-                      "use average_method='arithmetic' by default.",
-                      FutureWarning)
-        average_method = 'max'
+        average_method = 'arithmetic'
```
you should remove this one as well. If some tests are failing because of this new default, you should fix `average_method='max'` for these specific tests.

glemaitre commented on sklearn/metrics/cluster/supervised.py:
```diff
@@ -687,11 +687,10 @@ def adjusted_mutual_info_score(labels_true, labels_pred,
         are 'min', 'geometric', 'arithmetic', and 'max'.
-        If 'warn', 'max' will be used. The default will change to
-        'arithmetic' in version 0.22.
+        If 'warn', 'max' will be used.
```
Remove this line.

glemaitre reviewed (changes requested):
The issue is that you should just stop warning and put as a default the `average` now. You might need to change some tests which expected `'max'`

scouvreur committed:
Removed 'warn' from adjusted_mutual_info_score docstring and setting of 'warn' to 'max'

glemaitre commented on sklearn/metrics/cluster/tests/test_supervised.py:
```diff
@@ -223,15 +222,14 @@ def test_adjusted_mutual_info_score():
     # Adjusted mutual information
     ami = adjusted_mutual_info_score(labels_a, labels_b)
-    assert_almost_equal(ami, 0.27502, 5)
+    assert_almost_equal(ami, 0.27821, 5)
```
I comment here but this is for the decorator above. Could you remove the line

```python
@ignore_warnings(category=FutureWarning)
```

on the top of the test. The test should not raise anymore warning.

glemaitre commented:
Remove the explicit setting of `average_method='arithmetic'` in the following place:

* `plot_adjusted_for_chance_measures.py` -> l.59
* `plot_affinity_propagation.py` -> l.39
* `plot_dbscan.py` -> l.48
* `plot_kmeans_digits.py` -> l. 68

scouvreur commented:
Got it apologies, I actually found instances using :

```bash
ack "average_method='arithmetic'"
```

glemaitre commented:
Could you make the same change for the normalized_mutual_info_score (NMI). Since that it will have an impact on the same tests, it will be easier.

In addition, remove all `@ignore_warnings(category=FutureWarning)` in this file which are related to NMI and AMI.

scouvreur committed:
Removed instances where default average_method='arithmetic' was passed

scouvreur committed:
Removed 'warn' parameter from normalized_mutual_info_score as well

scouvreur committed:
Removed ignore future warning for test_adjusted_mutual_info_score as function no longer raises warning

glemaitre commented:
You are still missing the following lines t be removed:

* in `test_hierarchical.py` -> l. 131-132 / l. 241-242 / l. 304-305
* in `test_spectral_embedding.py` -> l. 89-90 / l. 184-185
* in `test_score_objects.py` -> l. 435

scouvreur committed:
Removed filter warnings in refs of ami or nmi

Removed pytest.mark.filterwarnings for deprecation of
parameters in v0.22 for functions in the codebase
calling adjusted_mutual_info_score or normalized_mutual_info_score.

The references were in:

   - test_hierarchical.py
   - test_spectral_embedding.py
   - test_score_objects.py

jnothman commented:
Thanks!
